File: w07/d03/Mason_Tang/hangman.js
function Hangman(word) {
  this.progress = function() {
    var guesses = [];

    word.split("").forEach(function(letter) {
      guesses.push("_");


    });

    return guesses.join(" ");
  }
  word_array = word.split("")

  this.guess = function(letter) {

    for(var i=0; i < word_array.length; i++)

    if(letter === word_array[i]) {
      console.log("true")
    } else {
      console.log("false")
    }



  };
}

var game = new Hangman("detective");


//this came first
console.log("_ _ _ _ _ _ _ _ _");
console.log(game.progress());


// console.log(false);
console.log(game.guess("e"));



// big picture then code. work out--> in

File: w10/d01/David_Montricher/hangman_david/lib/hangman.rb
class Hangman
  def initialize(word)
    @word = word
  end

  def progress()
    @word.split("").map {"_"}.join(" ")
  end

  def guess(letter)
    if @word.include?("letter")
      index = @word.index(letter)
      @progress[index] = letter
      return true
    else
      return false
    end
  end



end
